app/uk/gov/hmrc/agentclientrelationships/repository/InvitationsRepository.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientrelationships.repository
+
+import org.mongodb.scala.model.Filters.equal
+import org.mongodb.scala.model.Updates.{combine, set}
+import org.mongodb.scala.model._
+import play.api.Logging
+import uk.gov.hmrc.agentclientrelationships.config.AppConfig
+import uk.gov.hmrc.agentclientrelationships.model.{DeAuthorised, Invitation, InvitationStatus}
+import uk.gov.hmrc.agentmtdidentifiers.model.ClientIdentifier.ClientId
+import uk.gov.hmrc.agentmtdidentifiers.model.Service
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
+
+import java.time.{Instant, LocalDate}
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class InvitationsRepository @Inject() (mongoComponent: MongoComponent, appConfig: AppConfig)(implicit
+  ec: ExecutionContext
+) extends PlayMongoRepository[Invitation](
+      mongoComponent = mongoComponent,
+      collectionName = "invitations",
+      domainFormat = Invitation.format,
+      indexes = Seq(
+        IndexModel(
+          Indexes.ascending("created"),
+          IndexOptions()
+            .name("timeToLive")
+            .expireAfter(appConfig.invitationsTtl, TimeUnit.DAYS)
+        )
+      ),
+      replaceIndexes = true
+    )
+    with Logging {
+
+  def create(
+    arn: String,
+    service: Service,
+    clientId: ClientId,
+    suppliedClientId: ClientId,
+    clientName: String,
+    expiryDate: LocalDate
+  ): Future[Invitation] = {
+    val invitation = Invitation.createNew(arn, service, clientId, suppliedClientId, clientName, expiryDate)
+    collection.insertOne(invitation).toFuture().map(_ => invitation)
+  }
+
+  def findAllForAgent(arn: String): Future[Seq[Invitation]] =
+    collection.find(equal("arn", arn)).toFuture()
```

Comment:
It might be best to create indexes on the find fields - so `arn` and `invitationId` are used in these queries, we know we will need compound indexing for future queries but it's probably sensible to leave them until those queries are actually needed.

Reply:
I've just added a commit for this, let me know if any further tweaks needed 👍